### PR TITLE
Feature/confirm winners

### PR DESCRIPTION
**Pull request by dwthomas77**

### CODE REVIEW

File: src/actions/AuctionActions.js [hunk 1]
```diff
@@ -103,11 +125,46 @@ export function toggleAuctionDetail(auctionId) {
...
+        let objStr = `${auction.id}/winningBids`;
+        updateObj[objStr] = combineWinningBids(auction);
+    })
+    return updateObj;
+}
+
+function combineWinningBids(auction) {
+    
+    let winningBidsObj = auction.winningBids || {};
+    let winningBids = Object.keys(auction.bids).filter( bid => auction.bids[bid].checked );
```
Comment (evandana) [inline src/actions/AuctionActions.js]: `bid=> auction.bids[bid]`
should be
`bidId=> auction.bids[bidId]`

Comment (evandana) [inline src/actions/AuctionActions.js]: same across other cases of similar

File: src/components/confirmDialog/ConfirmDialog.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        <Dialog
+          title="Confirm Auction Winners"
+          actions={actions}
+          modal={true}
+          open={this.state.open}
+        >
+            <p>Are you sure you want to confirm the following auction winners?  Once confirmed you will
+            not be able to change the confirmation.</p>
+            
+            {awesomeBidListOfGreatness}
```
Comment (evandana) [inline src/components/confirmDialog/ConfirmDialog.js]: excellent var naming

File: src/actions/AuctionActions.js [hunk 3]
```diff
@@ -103,11 +125,46 @@ export function toggleAuctionDetail(auctionId) {
...
+            
+        newWinningBids.forEach( bidId => {
+            delete auction.bids[bidId].checked;
+            delete auction.bids[bidId].winner;
+            winningBidsObj[bidId] = auction.bids[bidId]
+        })
+
+    }
+
+    return winningBidsObj;
```
Comment (evandana) [inline src/actions/AuctionActions.js]: lots of iterating over all bids here. wonder if it's possible to improve on this logic to add back in "total earned" while also optimizing flow. perhaps by adding a prop onto the auction?

File: src/components/containers/header/Header.js [hunk 4]
```diff
@@ -78,9 +78,9 @@ class Header extends Component {
     }
 
     clearAuctionDetail(event) {
-        // TODO: Dave to hook up?
-        // const { dispatch } = this.props
-        // dispatch(clearAuctionDetail())
+        console.log("clearing auc details")
```
Comment (evandana) [inline src/components/containers/header/Header.js]: remove console.log

File: src/components/containers/header/Header.js [hunk 5]
```diff
@@ -78,9 +78,9 @@ class Header extends Component {
     }
 
     clearAuctionDetail(event) {
```
Comment (evandana) [inline src/components/containers/header/Header.js]: where does this get called from?

File: src/reducers/AuctionReducers.js [hunk 6]
```diff
@@ -8,14 +10,54 @@ import {
...
+
+            return Object.assign({}, state, {
+                ownedAuctionCollection: [
+                    ...state.ownedAuctionCollection
+                ],
+                confirmWinnersSubmitDisable: !hasCheckedNonWinners(state.ownedAuctionCollection)
+            });
+            
+        case CONFIRM_WINNERS:
+            console.log("Winners confirmed successfully");
```
Comment (evandana) [inline src/reducers/AuctionReducers.js]: remove console.log

File: src/reducers/AuctionReducers.js [hunk 7]
```diff
@@ -70,4 +171,106 @@ function auctions(state = defaultAuctionState, action) {
...
+                    if (bids[bidId].bidAmount > bids[lowestBid].bidAmount) {
+                        // console.log("but this bid is higher than lowest bid of "+bids[lowestBid].bidAmount+", so remove lowest and add this one")
+                        uniqueBids[uniqueBids.indexOf(lowestBid)] = bidId;
+                    } else {
+                        // console.log("lowest bid of "+bids[lowestBid].bidAmount+" is higher than this, ignore.")
+                    }
+                }
+            }
+        }
+    });
```
Comment (evandana) [inline src/reducers/AuctionReducers.js]: smart work here

File: src/reducers/AuctionReducers.js [hunk 8]
```diff
@@ -70,4 +171,106 @@ function auctions(state = defaultAuctionState, action) {
...
+    auction.bids = filterUniqueBidders(auction.bids);
+    // if (auction.bids) {
+    //     let firstItem = auction.bids[Object.keys(auction.bids)[0]];
+    //     firstItem.checked = true;
+    //     auction.bidTotal += firstItem.bidAmount;
+    // } 
+    if (auction.winningBids) {
+        Object.keys(auction.winningBids).forEach( bidId => {
+            auction.bidTotal += auction.bids[bidId].bidAmount;
+            auction.bids[bidId].checked = true;
```
Comment (evandana) [inline src/reducers/AuctionReducers.js]: i wonder if we can get away from tracking a prop on the bids. not sure

File: src/actions/AuctionActions.js [hunk 9]
```diff
@@ -24,6 +26,26 @@ export function auctionPushErrorHandler (error) {
...
+    };
+}
+
+export function confirmWinners () {
+    return (dispatch, getState) => {
+        let auctions = getState().auctions.ownedAuctionCollection;
+        let updateObj = buildConfirmUpdateObj(auctions);
+        firebase.updateWinningBids(updateObj)
+            .then( (msg) => {
+                console.log(msg);
```
Comment (evandana) [inline src/actions/AuctionActions.js]: remove console.log

File: src/actions/AuctionActions.js (hunk 3 above)
Comment (dwthomas77) [inline src/actions/AuctionActions.js]: I agree, the total sum logic can be refactored and we'll replace the total of all auctions.  After core functionality is merged into Dev I"ll continue improving this feature.

File: src/components/containers/header/Header.js (hunk 5 above)
Comment (dwthomas77) [inline src/components/containers/header/Header.js]: I fear this was a feature I started working on and then forgot about.  Will anyone be using the add auction form on game day or is that only controlled by you before the auction?

File: src/reducers/AuctionReducers.js (hunk 8 above)
Comment (dwthomas77) [inline src/reducers/AuctionReducers.js]: I'm totally open to discussion with this but I found managing view model data that isn't held in the store was very difficult, as when I update the props on a stateless functional component it won't actually re-render the component, in Redux only when the state which is mapped to the component changes is the component re-rendered.

**Pull request merged**